### PR TITLE
[cxxgraph] Add new port CXXGraph

### PR DESCRIPTION
**Pull request by QuantumFelidae**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by QuantumFelidae** [comment]
@microsoft-github-policy-service agree

File: ports/cxxgraph/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,13 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO ZigRazor/CXXGraph
+    REF v2.0.0
```
Comment (FrankXie05) [inline ports/cxxgraph/portfile.cmake:4]:
```suggestion
    REF "v{VERSION}"
```
Please use the new function.

File: ports/cxxgraph/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    SHA512 a4409c81132e6c7e34022c54d9a57b965970aa8e1fcd97b9f916334c1d480674a526e7d5ad727ab652e4842083249dea89de519b104c1f9f205423eabd3c2338
+    HEAD_REF master
+)
+
+vcpkg_cmake_configure(SOURCE_PATH "${SOURCE_PATH}")
+# cxxgraph provides no targets and is a header only lib designed to be copied to include dir
+file(INSTALL "${SOURCE_PATH}/include/" DESTINATION "${CURRENT_PACKAGES_DIR}/include/cxxgraph")
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/cxxgraph" RENAME copyright)
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
```
Comment (FrankXie05) [inline ports/cxxgraph/portfile.cmake:13]:
```suggestion
file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")

```
We need the end line.

File: ports/cxxgraph/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "cxxgraph",
+  "version": "2.0.0",
+  "description": "CXXGraph is a header only comprehensive C++ graph library.",
```
Comment (FrankXie05) [inline ports/cxxgraph/vcpkg.json:4]:
```suggestion
  "description": "CXXGraph is a header only comprehensive C++ graph library.",
  "homepage": "https://github.com/ZigRazor/CXXGraph"
```

File: ports/cxxgraph/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,13 @@
+vcpkg_from_github(
```
Comment (FrankXie05) [inline ports/cxxgraph/portfile.cmake:1]:
```suggestion
# Header-only library

vcpkg_from_github(
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Comment by FrankXie05** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". 
          That way, I can be aware that you've responded since you can't modify the tags.

**Comment by FrankXie05** [comment]
The usage tested successfully on `x64-windows`.

**Review by FrankXie05 — DISMISSED** [review]

File: ports/cxxgraph/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Header-only library
```
Comment (dg0yt) [inline ports/cxxgraph/portfile.cmake:1]: AFAIU there is no point in `vcpkg_cmake_configure` setting up the debug build type.
```suggestion
set(VCPKG_BUILD_TYPE release) # Header-only library
```

**Comment by QuantumFelidae** [comment]
The PR was not marked as draft, so consider this comment to be marking as ready for review :)

**Comment by BillyONeal** [comment]
Looks great, thanks for the new port!

**Pull request merged**